pkg/streamingpromql/aggregations/sum.go

```diff
@@ -89,34 +89,36 @@ func (g *SumAggregationGroup) accumulateHistograms(data types.InstantVectorSerie
 		g.histogramSums = g.histogramSums[:steps]
 	}
 
-	for _, p := range data.Histograms {
-		idx := (p.T - start) / interval
+	for inputIdx, p := range data.Histograms {
+		outputIdx := (p.T - start) / interval
 
-		if g.histogramSums[idx] == invalidCombinationOfHistograms {
+		if g.histogramSums[outputIdx] == invalidCombinationOfHistograms {
 			// We've already seen an invalid combination of histograms at this timestamp. Ignore this point.
 			continue
 		}
 
-		if g.histogramSums[idx] == nil {
+		if g.histogramSums[outputIdx] == nil {
+			data.Histograms[inputIdx].H = nil // Ensure the FloatHistogram instance is not reused when the HPoint slice is reused, as we'll retain a reference to the FloatHistogram instance below.
```

Comment:
Note to reviewers: this is the key line in this file, the rest of the diff is just renaming things